project/index.mjs

```diff
@@ -267,107 +261,99 @@ async function createNewProject(req, res) {
     res.status(201).json(logicResult)
     return
   } else {
-    utils.respondWithError(res, logicResult.status, logicResult.message)
+    respondWithError(res, logicResult.status, logicResult.message)
     return
   }
 }
 
-router
-  .route("/create")
-  .post(async (req, res, next) => {
-    // TODO: Add authentication to this endpoint
-    if (!utils.isValidJSON(req.body)) {
-      utils.respondWithError(res, 400, "Improperly formatted JSON")
-      return
-    }
-    await createNewProject(req, res)
-  })
-  .all((req, res, next) => {
-    utils.respondWithError(
-      res,
-      405,
-      "Improper request method, please use POST."
-    )
-  })
+router.route("/create").post(auth0Middleware(), async (req, res) => {
+  const user = req.user
 
-router.get("/:id", async (req, res, next) => {
-  let id = req.params.id
-  if (!database.isValidId(id)) {
-    utils.respondWithError(
-      res,
-      400,
-      "The TPEN3 project ID must be a hexadecimal string"
-    )
-    return
-  }
+  if (!user?.agent) return respondWithError(res, 401, "Unauthenticated user")
+
+  const projectObj = new Project(user?._id)
+
+  let project = req.body
+  project = {...project, creator: user?.agent}
 
   try {
-    const projectObj = await logic.findTheProjectByID(id)
-    if (projectObj) {
-      respondWithProject(req, res, projectObj)
-    } else {
-      utils.respondWithError(
-        res,
-        404,
-        `TPEN3 project "${req.params.id}" does not exist.`
-      )
-    }
-  } catch (err) {
-    utils.respondWithError(
+    const newProject = await projectObj.create(project)
+    res.status(200).json(newProject)
+  } catch (error) {
+    respondWithError(
       res,
-      500,
-      "The TPEN3 server encountered an internal error."
+      error.status || error.code || 500,
+      error.message || "Unknown server error"
```

Comment:
This might be safer as `??` but I'm not certain

Reply:
Pretty much the same thing. it checks in the other in which they appear.